qemu/tests/s390x_cpu_model_boot.py

```diff
@@ -0,0 +1,37 @@
+import json
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Boot guest with supported cpu-models
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    boot_cpu_models = params['boot_cpu_models'].split(';')
+    cpu_model_check_cmd = params.get('cpu_model_check_cmd')
+    cpu_model_check_args = json.loads(params.get('cpu_model_check_args'))
+    vm_name = params['main_vm']
+    vm = env.get_vm(vm_name)
+    host_model = vm.monitor.cmd(cpu_model_check_cmd,
+                                cpu_model_check_args).get('model')
+    host_model_name = host_model.get('name')[:-5]
+    for boot_models in boot_cpu_models:
+        if host_model_name in boot_models:
+            boot_cpu_models = boot_models.split(',')
```

Comment:
So during the `for` loop, `boot_cpu_models` will be override by `boot_cpu_models = boot_models.split(',')`, and you don't have break to exit the loop.

Reply:
solved